yt_dlp/extractor/bandcamp.py

```diff
@@ -436,45 +435,47 @@ class BandcampWeeklyIE(BandcampIE):  # XXX: Do not subclass from concrete IE
 
     def _real_extract(self, url):
         show_id = self._match_id(url)
-        webpage = self._download_webpage(url, show_id)
-
-        blob = self._extract_data_attr(webpage, show_id, 'blob')
-
-        show = blob['bcw_data'][show_id]
-
-        formats = []
-        for format_id, format_url in show['audio_stream'].items():
-            if not url_or_none(format_url):
-                continue
-            for known_ext in KNOWN_EXTENSIONS:
-                if known_ext in format_id:
-                    ext = known_ext
-                    break
-            else:
-                ext = None
-            formats.append({
-                'format_id': format_id,
-                'url': format_url,
-                'ext': ext,
-                'vcodec': 'none',
-            })
-
-        title = show.get('audio_title') or 'Bandcamp Weekly'
-        subtitle = show.get('subtitle')
-        if subtitle:
-            title += f' - {subtitle}'
+        show_data = self._download_json(
+            'https://bandcamp.com/api/bcradio_api/1/get_show',
+            show_id,
+            'Downloading Show Data.',
+            data=json.dumps({'id': show_id}).encode(),
+            headers={
+                'Content-Type': 'application/json',
+                'Origin': 'https://bandcamp.com',
+                'Referer': 'https://bandcamp.com/',
+            },
+        )
+        raw_metadata = self._extract_data_attr(
+            self._download_webpage(
+                url,
+                show_id,
+                'Downloading Show Metadata.',
+            ), show_id, 'blob',
+        )
+        metadata = traverse_obj(raw_metadata, (
+            'appData', 'shows', lambda _, mb: int(mb.get('showId')) == int(show_id),
+            any))
+
+        audio_data = show_data['radioShowAudio']
+        thum_id = metadata.get('imageId')
+        thum_url = f'https://f4.bcbits.com/img/000{thum_id}_171.jpg' if thum_id else None
```

Comment:
The leading zeroes do not appear to be necessary, and if we change `171` to `0` then we get the full size image

Import `format_field` from utils and do this:
```suggestion
        thumbnail = format_field(metadata, 'imageId', 'https://f4.bcbits.com/img/%s_0.jpg', default=None)
```
This can be inlined in the return below